cli/tools/lint.rs

```diff
@@ -42,21 +44,47 @@ fn create_reporter(kind: LintReporterKind) -> Box<dyn LintReporter + Send> {
 }
 
 pub async fn lint_files(
+  maybe_lint_config: Option<LintConfig>,
   args: Vec<PathBuf>,
   ignore: Vec<PathBuf>,
   json: bool,
 ) -> Result<(), AnyError> {
   if args.len() == 1 && args[0].to_string_lossy() == "-" {
-    return lint_stdin(json);
+    return lint_stdin(json, maybe_lint_config.as_ref());
   }
+
+  // Combine ignored files from config file and CLI flag.
+  let mut include_files = args;
+  let mut exclude_files = ignore;
+
+  if let Some(lint_config) = maybe_lint_config.as_ref() {
+    let include_files_from_config_file = lint_config
+      .files
+      .include
+      .iter()
+      .map(PathBuf::from)
+      .collect::<Vec<PathBuf>>();
+    include_files.extend(include_files_from_config_file);
+
+    let exclude_files_from_config_file = lint_config
+      .files
+      .exclude
+      .iter()
+      .map(PathBuf::from)
+      .collect::<Vec<PathBuf>>();
+    exclude_files.extend(exclude_files_from_config_file);
+  }
```

Comment:
I'm still debating whether we should "extend" default globs with what's listed in `lint.files.include` or should it override default glob (the same applies to combining `litn.files.exclude` and `--ignore` flag). 

@dsherret what are your thoughts?

Reply:
I'm still trying to figure this out in dprint and I'm not sure what the best way is. Right now I override.

Reply:
Thinking about it more I'm leaning towards higher precedence of CLI flags - ie. if you use `--ignore` or provide arguments to `deno lint` they are used instead of values from config file.

Example:
```
// Deno.jsonc
"lint": {
  "files": {
    "include": ["src/", "frontend/"],
    "exclude": ["src/testdata/"]
  },
  "rules": {
    "tags": ["recommended"],
    "include": ["ban-untagged-todo"]
  }
}

deno lint -c Deno.jsonc --ignore foo/ src/
```
Will use `rules` config as in `Deno.jsonc` but will ignore only `foo/` directory and include `src/` directory.